Models/global_config.hpp

```diff
@@ -70,29 +39,53 @@ namespace config
      
     //Define factory function to create motor drivers.
     //The driver class must be copy-constructable as the returned object will be a copy.
-    template <class BaseDriverType, class DerivedDriverType>
-    BaseDriverType constructDriver() {
-        DerivedDriverType driver;
-        return driver;
+    template <class BaseDriverType, class DerivedDriverType, auto... args>
+    BaseDriverType* constructDriver() {
+        return new DerivedDriverType(args...);
     }
```

Comment:
I tried that and got an error due to the driverConstructor field in the Motor_t struct not matching. The struct needs to use the general motor type because it doesn't know what specific type of motor constructor it will hold.

Reply:
I wonder if this is just a difference between how `std::function` and  the `type (*)(args)` format works. I'm fine to keep it either way.